Add docs for Structured Authn beta (#45108)
kubernetes/website: content/en/docs/reference/access-authn-authz/authentication.md

Author: I didn't see any mention of metrics in other section, so figured this isn't the right place to add this.

You must specify the path to the authentication configuration using the `--authentication-config` flag on the API server. If you want to use command line flags instead of the configuration file, those will continue to work as-is.
To access the new capabilities like configuring multiple authenticators, setting multiple audiences for an issuer, switch to using the configuration file.

For Kubernetes v{{< skew currentVersion >}}, the structured authentication configuration file format
is beta-level, and the mechanism for using that configuration is also beta. Provided you didn't specifically
disable the `StructuredAuthenticationConfiguration`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) for your cluster,
you can turn on structured authentication by specifying the `--authentication-config` command line
argument to the kube-apiserver. An example of the structured authentication configuration file is shown below.

{{< note >}}
If you specify `--authentication-config` along with any of the `--oidc-*` command line arguments, this is

> a misconfiguration. In this situation, the API server reports an errors and then immediately exits.

Reviewer: nit: error should be singular here

> ##### Using Authentication Configuration

Reviewer 2: I recommend
```suggestion
##### Authentication configuration from a file {#using-authentication-configuration}
```

{{< feature-state for_k8s_version="v1.30" state="beta" >}}

JWT Authenticator is an authenticator to authenticate Kubernetes users using JWT compliant tokens. The authenticator will attempt to
parse a raw ID token, verify it's been signed by the configured issuer. The public key to verify the signature is discovered from the issuer's public endpoint using OIDC discovery.

The minimum valid JWT payload must contain the following claims:

> ```json

Reviewer 3: this will fix syntax errors on comments

```suggestion
```yaml
```

{
  "iss": "https://example.com",   // must match the issuer.url
  "aud": ["my-app"],              // at least one of the entries in issuer.audiences must match the "aud" claim in presented JWTs.

>   "exp": 1234567890,              // token expiration time

Reviewer 4: ```suggestion
  "exp": 1234567890,              // token expiration as Unix time (the number of seconds elapsed since January 1, 1970 UTC)
```

  "<username-claim>": "user"      // this is the username claim configured in the claimMappings.username.claim or claimMappings.username.expression
}
```

The configuration file approach allows you to configure multiple JWT authenticators, each with a unique `issuer.url` and `issuer.discoveryURL`. The configuration file even allows you to specify [CEL](/docs/reference/using-api/cel/)

> expressions to map claims to user attributes, and to validate claims and user information. The API server also automatically reloads the authenticators when the configuration file is modified. An example configuration is provided below.

Author: Mentioned `apiserver_authentication_config_controller_automatic_reload_last_timestamp_seconds`

You must specify the path to the authentication configuration using the `--authentication-config` flag on the API server. If you want to keep using command line flags instead of the configuration file, those will continue to work as-is.

> To gain access to new capabilities like configuring multiple authenticators, setting multiple audiences for an issuer, switch to using the configuration file.

Reviewer 2: Avoid “gain”; this carries an implication that there is an existing cluster being upgraded. In fact, people might deploy a brand new cluster.

> Starting Kubernetes 1.30, the configuration file format is beta-level, and only requires specifying `--authentication-config` since the `StructuredAuthenticationConfiguration` feature gate is enabled by default.

Reviewer 4: ```suggestion
Starting Kubernetes v1.30, the configuration file format is beta-level, and only requires specifying `--authentication-config` since the `StructuredAuthenticationConfiguration` feature gate is enabled by default.
```

Reviewer 2: My take:
```suggestion
For Kubernetes {{< skew currentVersion }}, the structure authentication configuration file format
is beta-level, and the mechanism for using that configuration is also beta. Provided you didn't specifically
disable the `StructuredAuthenticationConfiguration`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) for your cluster,
you can turn on structured authentication by specifying the `--authentication-config` command line
argument to the kube-apiserver.
```

{{< note >}}

> If you want to keep using command line flags to configure JWT authenticator instead of a configuration file, those will continue to work as-is.

Reviewer 4: This sentence is repeated above?

Author: I wanted to emphasize that again in the note section but feels redundant. I've removed it from here.

> 

Reviewer 2: Avoid “keep”; this carries an implication that there is an existing cluster being upgraded. In fact, people might deploy a brand new cluster.

> Setting both `--authentication-config` and any of the `--oidc-*` flags will result in an error. If done, there will be an error and API server would exit right away.

Reviewer 2: ```suggestion
If you specify `--authentication-config` along with any of the `--oidc-*` command line arguments, this is
a misconfiguration. In this situation, the API server reports an errors and then immediately exits.
```

> If you want to use the feature, you have to remove the `--oidc-*` flags and use the configuration file instead.

Reviewer 2: ```suggestion
If you want to switch to using structured authentication configuration, you have to remove the `--oidc-*`
command line arguments, and use the configuration file instead.
```

  {{% /tab %}}
  {{< /tabs >}}

> ##### Limitations

Reviewer 4: ```suggestion
###### Limitations
```

Reviewer 2: That's a very deep heading; it suggests we could improve the way we lay things out.

1. Distributed claims do not work via [CEL](/docs/reference/using-api/cel/) expressions.
1. Egress selector configuration is not supported for calls to `issuer.url` and `issuer.discoveryURL`.

Kubernetes does not provide an OpenID Connect Identity Provider.

> You can use an existing public OpenID Connect Identity Provider (such as Google, or

Reviewer 4: The `Support OpenID connect discovery; not all do` bit below doesn't actually match the flexibility that `discoveryURL` provides?

Author: I've updated this. LMK what you think!
